src/network/networkd-route.c

```diff
@@ -510,6 +510,46 @@ static void route_copy(Route *dest, const Route *src, const MultipathRoute *m, c
         }
 }
 
+int route_dup(const Route *src, Route **ret) {
+        _cleanup_(route_freep) Route *dest = NULL;
+        MultipathRoute *m;
+        int r;
+
+        assert(src);
+        assert(ret);
+
+        r = route_new(&dest);
+        if (r < 0)
+                return r;
+
+        *dest = *src;
```

Comment:
Maybe
```C
    dest = new(Route, 1);
    if (!dest)
            return -ENOMEM;
    *dest = *src;
```

Right now we're setting the contents of `dest` twice without a good reason.

Reply:
OK. Now, `newdup()` is used.

Reply:
Oh, that's even nicer.